File: task9/src/classesFigure/Circle.java
package classesFigure;


import classesFigure.base.Figure;
import interfaceFigure.LengthCircle;


public class Circle extends Figure implements LengthCircle {
    private double length;

    public Circle(int radius) {
        if (isPositive(radius)) {
            super.name = "Окружность";
            this.length = 2 * Math.PI * radius;
            super.square = Math.PI * Math.pow(radius, 2);
        } else {
            super.name = "";
        }
    }

    @Override
    public double getLength() {
        return this.length;
    }

}

File: task13/base/Circle.py
from base.Figure import Figure
import math

class Circle(Figure):
    __name = "окружность"
    def __init__(self, radius):
        if (radius > 0):
            square = radius ** 2 * math.pi
            self.__length = 2 * math.pi * radius
            super().__init__(self.__name, square)
        else:
            print("Радиус меньше 0")


    def get_length(self):
        return self.__length

    def get_info(self):
        return f"{super().get_info()}, длина окр-ти: {self.__length}"
